### PR TITLE
UCT/IB/DC/VERBS: Make support of zero copy APIs in dc_verbs

### PR DESCRIPTION
**Pull request by shssf**
Related Issue #951
@shamisp Could you please take a look?

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2517/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1867/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2526/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2527/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1880/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1881/ for details (Mellanox internal link).

File: src/uct/ib/dc/verbs/dc_verbs.c [hunk 1]
```diff
@@ -148,22 +143,23 @@ static inline void uct_dc_verbs_iface_add_send_comp(uct_dc_verbs_iface_t *iface,
...
+                           size_t iovcnt, size_t iov_total_length, uint64_t remote_addr,
+                           uct_rkey_t rkey, uct_completion_t *comp, int opcode)
 {
     uct_dc_verbs_iface_t *iface = ucs_derived_of(ep->super.super.super.iface,
                                                  uct_dc_verbs_iface_t);
     struct ibv_exp_send_wr wr;
-    struct ibv_sge sge;
-
-    UCT_SKIP_ZERO_LENGTH(length);
+    struct ibv_sge sge[iovcnt];
```
Comment (yosefe) [inline src/uct/ib/dc/verbs/dc_verbs.c]: let's use UCT_IB_MAX_IOV or alloca for portability

File: src/uct/ib/dc/verbs/dc_verbs.c [hunk 2]
```diff
@@ -148,22 +143,23 @@ static inline void uct_dc_verbs_iface_add_send_comp(uct_dc_verbs_iface_t *iface,
...
+    size_t sge_cnt;
 
+    UCT_SKIP_ZERO_LENGTH(iov_total_length);
     UCT_DC_CHECK_RES(&iface->super, &ep->super);
-    UCT_RC_VERBS_FILL_RDMA_WR(wr, wr.exp_opcode, opcode, sge, length, remote_addr, rkey);
+
+    sge_cnt = uct_rc_verbs_rdma_zcopy_sge_fill_iov(sge, iov, iovcnt);
```
Comment (yosefe) [inline src/uct/ib/dc/verbs/dc_verbs.c]: we can check sge_cnt!=0 instead of UCT_SKIP_ZERO_LENGTH(iov_total_length);

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2564/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1916/ for details (Mellanox internal link).

**Pull request merged**